managed/src/SwiftlyS2.Core/Services/PluginConfigurationService.cs

```diff
@@ -96,21 +226,55 @@ public IPluginConfigurationService InitializeWithTemplate(string name, string te
     return this;
   }
 
-  public IPluginConfigurationService Configure(Action<IConfigurationBuilder> configure) {
+  public IPluginConfigurationService InitializeTomlWithModel<T>(string name, string sectionName) where T : class, new()
```

Comment:
Why not just make it generic? The only thing that differs from configuration files is how the file is created, nothing else. This is already a 3rd way to create a config and I'm sure people will love to add more formats in the future.

I suggest to deprecate the current two methods and create a generic one. In it, at the end, just do a switch statement and call private methods that would create the config. This would be a lot more maintainable, especially since I/O logic for all of these are the same